File: JAVA/S9/L1/stack2.java
class stack{
    int top;
    int size;
    int []arr;
    stack(int size)
    {
        this.size=size;
        top=-1;
        arr=new int[size];
    }
}
class s{
    int isFull(stack st)
    {
        if(st.top==st.size-1)return 1;
        return 0;
        
    }
    int isEmpty(stack st)
    {
        if(st.top==-1)return 1;
        return 0;
    }
    void push(int data,stack st)
    {
        if(isFull(st)==1)return ;
        st.arr[++st.top]=data;

    }
    int pop(stack st)
    {
        if(isEmpty(st)==1)return -1;
        return st.arr[st.top];

    }
    void display(stack st)
    {
        for(int i=0;i<=st.top;i++)
        {
            System.out.print(st.arr[i]+" ");
        }
    }
}
public class stack2
{
    public static void main(String[] args) {
        {
            int size=10;
            stack st=new stack(size);
            s stt=new s();
            stt.push(282,st);
            stt.push(272,st);
            stt.display(st);
        }
    }
}

File: C/S1_BASIC/L1_BASIC/adt/stack/static/stack2.c
//using structure
#include<stdio.h>
#include<stdlib.h>
struct stack{
    int capacity;
    int top;
    int *arr;
};
struct stack *createStack(int capacity){
    struct stack *s=(struct stack*)malloc(sizeof(struct stack));
    s->capacity=capacity;
    s->top=-1;
    s->arr=(int*)malloc(sizeof(int)*capacity);
    return s;

}
int isFull(struct stack *s){
    return s->top==s->capacity-1;
}
int isEmpty(struct stack *s){
    return s->top==-1;
}
void push(struct stack *s,int data){
    if(isFull(s)){
        printf("stack is full ,can not push the element");
        return;
    }
    s->arr[++(s->top)]=data;
}
int pop(struct stack *s){
    if(isEmpty(s)){
        printf("stack is empty ,can not remove element");
        return -1;
    }
    return s->arr[(s->top)--];
}
void display(struct stack *s){
    if(isEmpty(s)){
        printf("stack is empty ,can not display element\n");
        return;
    }
    for(int i=0;i<=s->top;i++){
        printf("%d ",s->arr[i]);
    }
}
int main(){
    struct stcak *s=createStack(10);
    push(s,56);
    push(s,89);
    display(s);
    return 0;

}